src/rgw/rgw_lua_background.cc

```diff
@@ -124,21 +148,9 @@ const BackgroundMapValue& Background::get_table_value(const std::string& key) co
 void Background::run() {
   ceph_pthread_setname("lua_background");
   const DoutPrefixProvider* const dpp = &dp;
-  lua_state_guard lguard(cct->_conf->rgw_lua_max_memory_per_state,
-                         cct->_conf->rgw_lua_max_runtime_per_state, dpp);
-  auto L = lguard.get();
+  std::unique_ptr<lua_state_guard> lguard = initialize_lguard_state();
+  auto L = lguard->get();
```

Comment:
you can check the unique pointer itself. it has a boolean casting operator:
https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool
so you can just write:
```cpp
if (!lguard) {
  return;
}
```
same in line 175
```